### PR TITLE
Made Payments page query based

### LINKED ISSUE #967: Invite-links form is broken when <3 group members.
### Problem

Under Group Settings -> Invite Links -> "propose adding a new member", the "Add new members" form fails when there are fewer than 3 members.

1. There is no visual feedback that it failed.
2. In devtools, this error shows up:

     | (anonymous) | @ | captureLogs.js:157
   -- | -- | -- | --
     | submit | @ | AddMembers.vue?rollu…n-vue=script.js:136
     | invokeWithErrorHandling | @ | vue.esm.js:1863
     | invoker | @ | vue.esm.js:2188
     | submit | @ | ProposalTemplate.vue…n-vue=script.js:207
     | invokeWithErrorHandling | @ | vue.esm.js:1863
     | invoker | @ | vue.esm.js:2188
     | submit | @ | ButtonSubmit.vue?rol…in-vue=script.js:65
     | invokeWithErrorHandling | @ | vue.esm.js:1863
     | invoker | @ | vue.esm.js:2188
     | original._wrapper | @ | vue.esm.js:7565

   It makes sense that `form.reason` on line 126 is `null` since that field doesn't exist in the form:

   Screen Shot 2020-09-03 at 11 04 39 AM
3. ~~It also seems impossible to invite someone with only a full name, not an email.~~ (The third point can be ignored, since I misunderstood this: the name is only to identify the invitation in the site UI, but the link is meant to be shared manually rather than automatically emailed.)

### Solution

1. When an error occurs, the UI should show some feedback, even if it's a generic error message like "Oops, something went wrong, sorry. Let us know at [email]".
2. Either the reason field should be added, or the line should be removed.
3. ~~I think there needs to be an "email" field too.~~

### LINKED ISSUE #993: Payment page's tabs should be query-based
### Problem

Going to the `Sent` page and refreshing will end up back on the `Todo` page within the Payments page.

### Solution

Per #935 if this uses query params to choose what tab to show, refreshing/deep-linking will be supported.

### LINKED ISSUE #1081: Error in Cypress tests: Cannot read property 'incomeDetailsType' of undefined
### Problem

Periodically (perhaps not on every run) the following error will appear in `master` in the Cypress console when running the `group-contributions.spec.js` tests:

```
captureLogs.js:157 uncaught Vue error in render: TypeError: Cannot read property 'incomeDetailsType' of undefined
    at Proxy.__vue_render__$2 (Contributions.vue:3)
    at VueComponent.Vue._render (vue.esm.js:3557)
    at VueComponent.updateComponent (vue.esm.js:4075)
    at Watcher.get (vue.esm.js:4488)
    at Watcher.run (vue.esm.js:4563)
    at flushSchedulerQueue (vue.esm.js:4319)
    at Array.<anonymous> (vue.esm.js:1989)
    at flushCallbacks (vue.esm.js:1915)
```

We've tracked this bug down to access of `ourGroupProfile` within `Contributions.vue` during specific moments when that returns undefined because this getter in `group.js` is allowed to return undefined and we do not check for that:

```js
    groupProfile (state, getters) {
      return username => {
        const profiles = getters.currentGroupState.profiles
        return profiles && profiles[username]
      }
    },
```

### Solution

There are two solutions to this bug. One is to make `groupProfile` return an empty object for a non-existent profile, e.g. something like this:

```js
    groupProfile (state, getters) {
      return username => {
        const profiles = getters.currentGroupState.profiles
        return profiles && profiles[username] || {}
      }
    },
```

However, that's not super clean. It makes sense to return undefined for a profile that doesn't exist. So perhaps a better solution is to handle this undefined case in `Contributions.vue` somehow.

There are several unchecked accesses of `ourGroupProfile` in `Contributions.vue`:

- Line 5: `add-income-details-widget(v-if='!ourGroupProfile.incomeDetailsType')`
- Line 181: `const amount = this.ourGroupProfile[this.ourGroupProfile.incomeDetailsType]`
- Line 186: `return this.ourGroupProfile.incomeDetailsType === 'incomeAmount'`

If we add checks for `undefined` here the error should go away.

### PR DESCRIPTION
**Pull request by Silver-IT**
### Summary of Changes

- Closes #993 
- Closes #1081
- Closes #967

### CODE REVIEW

**Review by Silver-IT — COMMENTED** [review]
@taoeffect, @SebinSong. Seems ready for your reviews.

File: frontend/views/containers/group-settings/InvitationsTable.vue [hunk 1]
```diff
@@ -275,7 +277,11 @@ export default ({
     },
     handleInviteClick (e) {
       if (e.target.classList.contains('js-btnInvite')) {
-        sbp('gi.actions/group/checkGroupSizeAndProposeMember', { contractID: this.currentGroupId })
+        if (this.groupShouldPropose) {
+          sbp('gi.actions/group/checkGroupSizeAndProposeMember', { contractID: this.currentGroupId })
+        } else {
+          sbp('okTurtles.events/emit', OPEN_MODAL, 'InvitationLinkModal')
+        }
```
Comment (SebinSong) [inline frontend/views/containers/group-settings/InvitationsTable.vue:284]: Nice spotting the bug and fix here.

File: frontend/views/pages/Payments.vue [hunk 2]
```diff
@@ -213,6 +214,21 @@ export default ({
...
+    } else {
+      const defaultTab = possibleSections[0]
+      if (defaultTab) {
+        this.handleTabClick(defaultTab)
+      } else if (section) {
+        const query = omit(this.$route.query, ['section'])
+        this.$router.push({ query }).catch(logExceptNavigationDuplicated)
+      }
+    }
+  },
```
Comment (SebinSong) [inline frontend/views/pages/Payments.vue]: @Silver-IT it looks like the logic in this `mounted()` hook is more or less duplicate of what `$route` watcher below does. if the intention here is taking care of the same logic upon the first render of the component, I _think_ `immediate: true` flag in the watcher can be used instead. (reference in [PageSection.vue](https://github.com/okTurtles/group-income/blob/5f489c1fcc721d79ecebe688fbf27d49130396dd/frontend/views/components/PageSection.vue#L32C1-L41C6))

**Comment by SebinSong** [comment]
Looks good to me @Silver-IT other than one comment I left above. Great work.

File: frontend/views/pages/Payments.vue (hunk 2 above)
Comment (Silver-IT) [reply frontend/views/pages/Payments.vue]: Thank you, @SebinSong. I never knew that before.

**Review by taoeffect — APPROVED** [review]
Nicely done @Silver-IT! Approved!

**Pull request merged**